src/main/java/Duke.java

```diff
@@ -1,10 +1,128 @@
 public class Duke {
+
+    private static TaskList taskList = new TaskList();
+
+    private static PersistentStorageHandler persistentStorageHandler = new PersistentStorageHandler();
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        UserInterface.printWelcome();
+
+
+
+        try {
+            if (persistentStorageHandler.ensureTaskFileExists()) {
+                taskList = persistentStorageHandler.readTaskFileFromDisc();
+                int numTasks = taskList.getNumberTasks();
+                UserInterface.print("Read existing tasks (" + numTasks + ") from disc");
+            }
+        } catch (DukeException e) {
+            UserInterface.showError(e.getMessage());
+        }
+
+        boolean isExit = false;
+        while(!isExit) {
+            try {
+                String userInput = UserInterface.getUserInput();
+                CommandType commandType = CommandParser.parseCommand(userInput);
+
+                switch (commandType) {
+                    case LIST:
```

Comment:
Should the indentation of the case clauses follow the indentation of the switch statement?